File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        ArrayList<Integer> stack = new ArrayList<>();

        for (int i = 0; i < N; i++) {
            String command = br.readLine();

            if (command.startsWith("push")) {
                int X = Integer.parseInt(command.split(" ")[1]);
                stack.add(X);

            } else if (command.equals("pop")) {
                if (!stack.isEmpty()) {
                    System.out.println(stack.get(stack.size() - 1));
                    stack.remove(stack.size() - 1);
                } else {
                    System.out.println(-1);
                }

            } else if (command.equals("size")) {
                System.out.println(stack.size());

            } else if (command.equals("empty")) {
                System.out.println(stack.isEmpty() ? 1 : 0);

            } else if (command.equals("top")) {
                if (!stack.isEmpty()) {
                    System.out.println(stack.get(stack.size() - 1));
                } else {
                    System.out.println(-1);
                }
            }



        }
    }


}

File: 백준/Silver/10828. 스택/스택.py
from collections import deque

import sys
input = sys.stdin.readline

def main():
    n = int(input())
    stack = deque()
    for _ in range(n):
        line = input().split()
        com = line[0]

        if com == "push":
            i = int(line[1])
            stack.append(i)

        elif com == "pop":
            if stack:
                print(stack.pop())
            else:
                print(-1)

        elif com == "size":
            print(len(stack))

        elif com == "empty":
            if stack:
                print(0)
            else:
                print(1)

        elif com == "top":
            if stack:
                print(stack[-1])
            else:
                print(-1)

if __name__ == "__main__":
    main()